### PR TITLE
improve the deploy scripts

### PR DESCRIPTION
**Pull request by viv-defy**

### CODE REVIEW

File: contracts/script/deploy_local.sh [hunk 1]
```diff
@@ -1,13 +1,36 @@
...
-until curl -s http://localhost:8545 >/dev/null 2>&1; 
+# loop to check if anvil has started
+until curl -s http://localhost:8545 >/dev/null;
 do
     sleep 1
 done
 
+# the following set of lines enable us to deploy the script from anywhere in the project
+# shellcheck disable=SC2046
+# shellcheck disable=SC2164
```
Comment (ska1296) [inline contracts/script/deploy_local.sh:12]: ??

File: contracts/script/deploy_local.sh [hunk 2]
```diff
@@ -1,13 +1,36 @@
...
 cd $(dirname "$0")
-source ../.env
-CREATE2_ADDRESS="0x4e59b44847b379578588920cA78FbF26c0B4956C"
-curl http://localhost:8545 -X POST -H 'Content-Type: application/json' --data "{\"jsonrpc\":\"2.0\", \"id\":1, \"method\": \"anvil_setCode\", \"params\": [\"$CREATE2_ADDRESS\", \"0x7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffe03601600081602082378035828234f58015156039578182fd5b8082525050506014600cf3\"]}"
-forge script DeployLocal.s.sol:DeployLocal --rpc-url $RPC_URL --broadcast
\ No newline at end of file
+# shellcheck disable=SC2046
+# shellcheck disable=SC2164
+cd $(dirname $(pwd))
```
Comment (ska1296) [inline contracts/script/deploy_local.sh:16]: do you need both? pwd and $0

File: contracts/script/deploy_local.sh (hunk 1 above)
Comment (viv-defy) [reply contracts/script/deploy_local.sh:12]: The IDE was throwing up warnings. So I used these shell checks, suggested by the IDE, for suppressing the warnings

File: contracts/script/deploy_local.sh (hunk 2 above)
Comment (viv-defy) [reply contracts/script/deploy_local.sh:16]: $0 gives the path to the deploy script, from the directory that the bash command is run from. We can then use it to move to the parent directory.
But, this is an issue when we run it from contracts/scripts dir. $0 gives us `.`. We cannot move to the parent directory of `.`. So, we have to use pwd to get the full path

File: contracts/script/deploy_local.sh (hunk 1 above)
Comment (ska1296) [reply contracts/script/deploy_local.sh:12]: don't suppress warning. address them.

File: contracts/script/deploy_local.sh [hunk 3]
```diff
@@ -1,13 +1,37 @@
...
+source .env
+CREATE2_ADDRESS="0x4e59b44847b379578588920cA78FbF26c0B4956C" # a fixed address at which anvil checks for the create 2 factory
+curl http://localhost:8545 -X POST -H 'Content-Type: application/json' --data "{\"jsonrpc\":\"2.0\", \"id\":1, \"method\": \"anvil_setCode\", \"params\": [\"$CREATE2_ADDRESS\", \"0x7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffe03601600081602082378035828234f58015156039578182fd5b8082525050506014600cf3\"]}" >/dev/null 2>&1;
+
+var=$(forge script script/DeployLocal.s.sol:DeployLocal --rpc-url "${RPC_URL}" --broadcast 2>&1) # store the outputs/logs of forge scrip to the variable
+
+# loop to check for errors
+while IFS= read -r line; do
+  if [[ "$line" == "Error:"* ]]; then
```
Comment (ska1296) [inline contracts/script/deploy_local.sh:25]: I always get this error irrespective of anvil's status. This is misleading.

**Review by ska1296 — CHANGES_REQUESTED** [review]

**Review by ska1296 — APPROVED** [review]

**Pull request merged**